src/Magnum/GL/Test/BufferImageGLTest.cpp

```diff
@@ -33,6 +33,10 @@
 #include "Magnum/GL/PixelFormat.h"
 #include "Magnum/GL/OpenGLTester.h"
 
+#ifndef MAGNUM_TARGET_GLES2
+#include "Magnum/DebugTools/BufferData.h"
```

Comment:
Ensure this library is also linked in everywhere except ES2 -- which is the actual root cause for the TODO you have ;)

> On some CIs DebugTools/BufferData.cpp isn't being compiled, leading to missing symbols. Investigate.

Currently because it's inlined on WebGL and desktop, you don't have a linker failure there, but with the anticipated template removal it'll be deinlined everywhere, so just link it everywhere except ES2.

Same for the other three tests.